resources/lib/plugin.py

```diff
@@ -179,6 +184,495 @@ def result_to_itemlist(result, playlist='', channel=''):
 
     return items
 
+def get_user_channel():
+    user_channel_str = ADDON.getSettingString('user_channel')
+    if user_channel_str:
+        toks = user_channel_str.split('#')
+        if len(toks) == 2:
+            return (toks[0], toks[1])
+        else:
+            ADDON.setSettingString('user_channel', '')
```

Comment:
nuking whatever users enter here manually, even if not correct, is not making us many friends :)